hraven-core/src/main/java/com/twitter/hraven/rest/client/UrlDataLoader.java

```diff
@@ -20,40 +24,63 @@
     private TypeReference typeRef;
     private int connectTimeout;
     private int readTimeout;
+    private boolean useCompression;
 
     /**
-     * Constructor.
+     * Constructor, defaults to using compression (gzip / deflate).
      * @param endpointUrl
      * @param t TypeReference for json deserialization, should be TypeReference<List<T>>.
      * @throws java.io.IOException
      */
     public UrlDataLoader(String endpointUrl, TypeReference t, int connectTimeout, int readTimeout)
         throws IOException {
+      this(endpointUrl, t, connectTimeout, readTimeout, true);
+    }
+
+    /**
+     * Constructor.
+     * @param endpointUrl
+     * @param t TypeReference for json deserialization, should be TypeReference<List<T>>.
+     * @throws java.io.IOException
+    */
+    public UrlDataLoader(String endpointUrl, TypeReference t, int connectTimeout, int readTimeout,
+                         boolean useCompression) throws IOException {
       this.endpointURL = endpointUrl;
       this.typeRef = t;
       this.connectTimeout = connectTimeout;
       this.readTimeout = readTimeout;
+      this.useCompression = useCompression;
     }
 
     @SuppressWarnings("unchecked")
     public List<T> load() throws IOException {
       InputStream input = null;
+
+      RequestConfig requestConfig =
+          RequestConfig.custom()
+              .setConnectTimeout(connectTimeout)
+              .setConnectionRequestTimeout(connectTimeout)
+              .setSocketTimeout(readTimeout).build();
+      HttpClientBuilder httpClientBuilder =
+          HttpClientBuilder.create().setDefaultRequestConfig(requestConfig);
+
+      if (! useCompression) {
+        LOG.info("Not using compression!");
+        httpClientBuilder.disableContentCompression();
+      } else {
+        LOG.info("Using compression by default! Trying gzip, deflate");
```

Comment:
Yes, chatted with Piyush offline. We can change the default one using compression to debug mode and keep the "not using compression" in info mode